roles/orasw_download_patches/tasks/main.yml

```diff
@@ -0,0 +1,127 @@
+---
+- name: Download patches from Oracle on ansible-controller just once
+  run_once: true
+  delegate_to: "{{ oracle_patch_download_host }}"
+  block:
+    - name: Check if credentials are known
+      ansible.builtin.assert:
+        that:
+          - mos_login is defined
+          - mos_password is defined
+          - oracle_sw_source_local is defined
```

Comment:
The assert is wrong. replace `oracle_sw_source_local` with `oracle_patch_download_dir`. Otherwise the default could not be set without defining `oracle_sw_source_local`